File: bernoulli/src/main/java/co/kruzr/bernoulli/managers/GPSManager.java
package co.kruzr.bernoulli.managers;

import android.content.Context;
import android.provider.Settings;

import lombok.AllArgsConstructor;

/**
 * A class to perform GPS related checks.
 */
@AllArgsConstructor
public class GPSManager {

    private final Context context;

    /**
     * @return  true if the GPS is ON, else false
     */
    public boolean isGPSTurnedOn() {

        boolean returnValue = false;

        int locationMode = 0;

        try {
            locationMode = Settings.Secure.getInt(context.getContentResolver(), Settings.Secure.LOCATION_MODE);

        } catch (Settings.SettingNotFoundException e) {
            e.printStackTrace();
            returnValue = false;
        }

        if (locationMode != Settings.Secure.LOCATION_MODE_OFF)
            returnValue = true;

        return returnValue;
    }
}

File: bernoulli/src/main/java/co/kruzr/bernoulli/managers/GPSManager.kt
package co.kruzr.bernoulli.managers

import android.content.Context
import android.provider.Settings
import android.provider.Settings.SettingNotFoundException

/**
 * A class to perform GPS related checks.
 */
internal class GPSManager constructor(private val context: Context){

    /**
     * @return true if the GPS is ON, else false
     */
    val isGPSTurnedOn: Boolean

        get() {

            var returnValue = false
            var locationMode = 0

            try {
                locationMode = Settings.Secure.getInt(context.contentResolver,
                        Settings.Secure.LOCATION_MODE)
            } catch (e: SettingNotFoundException) {
                e.printStackTrace()
                returnValue = false
            }

            if (locationMode != Settings.Secure.LOCATION_MODE_OFF) returnValue = true

            return returnValue
        }
}